### ISSUE TITLE
Carb Pool names & pool-based IC files

### ISSUE THREAD

**Issue by mdietze** [PecanProject/pecan#1442]
Had a chat with @istfer and @mccabete and wanted to bring everyone onto the same page (especially @annethomas )

1) SIPNET: We decided that while setting up a non-default param file that modifies the IC is a fine short term solution for testing, we do want to go with having an explicit IC file

2) We decided that it makes sense to have a standard file format for pool-based models. As we’ve discussed previously, I’d advocate netCDF. 

* For data that’s cohort-based, we’ll have to add a function to the IC workflow that aggregates up to these pools
   + We should decide on the names/units of these pools
   + It would be awfully helpful if those names/units are the same ones we decide to use for expanding the output file format (per discussion with Tess about the CarbPools variable)
* For data that's already pool-based, we'll want to have functions that convert data from it's current format into this standard, for these functions to leverage load_data if possible, and to work these functions into the IC workflow eventually
* For models that have custom file formats for such things, we'll need their IC conversion functions to go from this standard to model-specific
* For models that lack a dedicated IC file, but rather munge that info in with other files (e.g. SIPNET, DALEC) we'll pass the std netCDF file to the model-specific write.config.
    + this has the advantage that many of these pools may need updating/converting with run-specific SLA, C:N, C content, etc parameters
* Right now we see no reason this workflow won't support ensemble-based uncertainty propagation

OK, the first discrete task so others can move forward: std variable names and units

**Comment by mccabete** [comment]
@mdietze What cohort based models should I be thinking about for the CarbPools aside from ED? I just want to get a sense of common pools that get used.

**Comment by mdietze** [comment]
@mccabete I don't think we want different CarbPools for the cohort-based models. CarbPools is aggregated data.  The issue of what cohort-based variables we need to add to the overall output standard is largely separate from the issue of defining the CarbPool names & units.

**Comment by mccabete** [comment]
@mdietze OK.  What is CarbPools used for then? We have the largest pools already represented in the mstmip standard. Do we want smaller pools? If it's not soil carbon, living biomass, respiration etc, what are we looking for and how long until it becomes cohort dependent?

**Comment by mdietze** [comment]
@mccabete Not 'smaller' pools, but rather the disaggregation of those pools. For example, splitting living biomass into leaves, stems, roots, reproduction, storage, etc and splitting soil carbon into the model pools.

**Comment by mccabete** [comment]
@mdietze Do you have some models in mind that use these kinds of pools?

**Comment by mdietze** [comment]
All of them have these pools, though they differ in how they split things up. Which raises an interesting question of how we want to balance lumping and splitting and what a model is supposed to do if we define 'standard' variables that they can't disaggregate.  But, on the other hand, we may not want to go to the lowest common denominator as that will cause us to loose info. On the other hand, we could define a hierarchy of pools and pool names, and models would just report the finest they can split things, but the hierarchy would give us a rule for aggregating things automatically so we can make model-model comparisons.

**Comment by mccabete** [comment]
@mdietze I like the hierarchy idea. That lends itself to cohort variables down the line, because we could report the same variables but treat the different cohorts as a subset. 

Could we implement it so that it automatically creates a nested object within the Carbpools variable or would it have to have the same structure as the other variables?

**Comment by mdietze** [comment]
I don't think we want to try and store a nested hierarchical object in netCDF, nor would it be efficient since it would result in the storage of a lot of redundant information.

We may or may may not recycle these new variable names at the cohort level, but we definitely won't change the definition of Carbpools to account for cohort structure, we'd create new variables that have explicit cohort dimensions.

**Comment by dlebauer** [comment]
One of the features of netCDF 4 is its support for hierarchical data (
http://unidata.github.io/netcdf4-python/#section2) through the use of
groups. Not sure if this is helpful for this application, e.g. using groups
for cohorts, but just wanted to make sure you are aware that this feature
exists.

**Comment by mccabete** [comment]
This is really tied to issue #1487 (which is a subset of this issue) 

Here's a first pass at making this happen just using MSTMIP variables and the model outputs that are summed into MSTMIP variables in model2netcdf.  

[Hierarchical structure of variable names .pdf](https://github.com/PecanProject/pecan/files/1099019/Hierarchical.structure.of.variable.names.pdf)
[Here's also a link to an online version, but you have to log into Lucid Chart.](https://www.lucidchart.com/invitations/accept/29a89965-d3a6-4455-8edb-6adee53d9963) 

Second step would be: Expanding this hierarchy to include all of the variables that can be reported from models (For FATES + CLM that is a little intimidating) 

Insuring that BETY variables are either included in this hierarchy or a removed for being redundant.

**Comment by mdietze** [comment]
@mccabete Noticed that you closed this, but AFAIK this hasn't been fully implemented yet
